### ISSUE TITLE
[v6] toString() method with makeObservable()

### ISSUE THREAD

**Issue by BlackRockSoul** [mobxjs/mobx#2485]
I'm unable to have `toString()` method in a class with `makeObservable()` at the same time. Receive `Types of property 'toString` are incompatible`

**Intended outcome:**

Class `Color` with custom `toString()` method with no errors

**Actual outcome:**

```
TS2345: Argument of type 
'{ _r: IObservableFactory; r: IComputedFactory; : IObservableFactory; g: IComputedFactory; : IObservableFactory; b: IComputedFactory; : IObservableFactory; a: IComputedFactory; }' 
is not assignable to parameter of type 'AnnotationsMap<Color, "_r" | "_g" | "_b" | "_a">'.  
Type '{ _r: IObservableFactory; r: IComputedFactory; _g: IObservableFactory; g: IComputedFactory; _b: IObservableFactory; b: IComputedFactory; _a: IObservableFactory; a: IComputedFactory; }' 
is not assignable to type '{ readonly r?: boolean | Annotation | undefined; readonly g?: boolean | Annotation | undefined; readonly b?: boolean | Annotation | undefined; readonly a?: boolean | ... 1 more ... | undefined; toString?: boolean | ... 1 more ... | undefined; asBackground?: boolean | ... 1 more ... | undefined; setColor?: boolean | ...'.     

Types of property 'toString' are incompatible.
      Type '() => string' is not assignable to type 'boolean | Annotation | undefined'.
         Property 'annotationType_' is missing in type '() => string' but required in type 'Annotation'.  annotation.d.ts(2, 5): 'annotationType_' is declared here.
```

**How to reproduce the issue:**

Create a class with the public `toString()` method and some properties.
In constructor write `makeObservable(this, {...some annotations})`

[CodeSandbox](https://codesandbox.io/s/magical-bush-n22rt?file=/index.ts)

**Versions**

```
"mobx": "6.0.0",
"typescript": "4.0.3"
```

**Comment by mweststrate** [comment]
Thanks for the detailed report! Releasing a fix as 6.0.1

**Comment by BlackRockSoul** [comment]
Just found an easy temporary solution. To define `toString` as a private method

```typescript
    makeObservable<Color, "_r" | "_g" | "_b" | "_a" | "toString">(this, {
      _r: observable,
      r: computed,
      _g: observable,
      g: computed,
      _b: observable,
      b: computed,
      _a: observable,
      a: computed,
      toString: false,
    });
```

[CodeSandbox](https://codesandbox.io/s/wandering-silence-wl011?file=/index.ts)
